CMakeLists.txt

```
@@ -26,13 +26,13 @@ INCLUDE(cmake/boost.cmake)
 # Project definition
 COMPUTE_PROJECT_ARGS(PROJECT_ARGS LANGUAGES CXX)
 PROJECT(${PROJECT_NAME} ${PROJECT_ARGS})
+CHECK_MINIMAL_CXX_STANDARD(11 REQUIRED)
 
 INCLUDE(cmake/pthread.cmake)  # needs to be included after the CXX definition
 
 # Project dependencies
 ADD_PROJECT_DEPENDENCY(Eigen3 REQUIRED PKG_CONFIG_REQUIRES eigen3)
-SET(BOOST_COMPONENTS serialization system unit_test_framework thread)
-SEARCH_FOR_BOOST()
+ADD_PROJECT_DEPEDENCY(Boost REQUIRED COMPONENTS serialization system unit_test_framework thread)
```

maybe this is better ?
```suggestion
ADD_PROJECT_DEPEDENCY(Boost REQUIRED COMPONENTS serialization system thread)
find_package(Boost REQUIRED COMPONENTS unit_test_framework)
```

---

I thought at it, but don't really see benefits, so at first I was too lazy to do the clean thing. But on a second thought, this would be a great way to just check if everything goes well when we have both find_package and ADD_PROJECT_DEPEDENCY side by side.

---

It also avoids propagating bad examples :)

---

It was a good idea to test that: 40ae6cc5698d18c5ef55b507d027f310af8c07fa

---

> I suggest to use ${Boost_LIBRARIES} for exported ones, and eg.
${Boost_UNIT_TEST_FRAMEWORK_LIBRARY} for the private ones.

You could also use the target directly and not the opaque variables.

---

Do you mean `Boost::unit_test_framework` ? This would be great, I'll check if this works everywhere, thanks.

---

`Boost::component` for all the component you need, not only UTF.